### DISCUSSION TITLE
Trying to programmatically pair with a BLE FIDO token that requires a passkey

### DISCUSSION THREAD

**Discussion by fishstiqz** [hbldh/bleak#767] [Q&A]
Hello. Bleak looks like a great library so far in my testing. However I cannot seem to figure out how to pair with my BLE device. My device is a BLE FIDO token that requires a passkey (6 digit code written on the back of the device itself) in order to pair. How does one supply this passkey using bleak and complete the pairing?

**Comment by dlech** [comment]
Pairing currently uses the OS interface for entering the PIN/passphrase (the OS should pop up a dialog box asking for it). There is a pull request open to allow programmatically entering the PIN/passphrase, but this needs work before being merged. Also, it will only be possible on Windows and Linux, not other OSes.

**Reply by fishstiqz** [reply]
Ok. I am excited about this PR. This is for a headless linux system so those OS restrictions are fine.

**Comment by mr-manuel** [comment]
Is there any update on entering the pairing PIN trought the script?

**Reply by bojanpotocnik** [reply]
Majority of the work has been done on [pairing-agent](https://github.com/hbldh/bleak/tree/pairing-agent) branch, but it has been stalled. I have also done some additional work within https://github.com/hbldh/bleak/pull/1133, however I am also waiting on what is going to happen with `pairing-agent` to continue.

**Reply by mr-manuel** [reply]
Thanks

**Comment by Jesse-AH-Carpenter** [comment]
Have there been any updates in regards to pairing using bleak. I would like to use LESC Just works, so no pin is needed.
